learning-progression.md

```diff
@@ -239,14 +239,30 @@ basic.forever(function () {
     basic.showIcon(IconNames.Heart)
     basic.pause(2000)
     basic.clearScreen()
+    basic.pause(2000)
 })
 ```
 
 #### 1. Study
+It is common to repeat code that is already writen. This is done with loops. One of the ways that this is done is with the [forever loops.]{https://makecode.microbit.org/reference/basic/forever}
```

Comment:
1. You first need to talk about _first-class functions_. This may open Pandora's box, so be careful. Stay minimal but technically correct.  
2. Then zoom in on functions _as arguments_ to other functions. This is tricky. Give it a shot and we'll triage it.  
3. Then explain how `forever` is a function that mimics a `while (true)` loop, taking a function as an argument instead of putting it in a block, as in the `while (true)` loop.